pubpub/pubpub
# Prevent archiving empty discussions + sentry/sourcemap fixes

## Issue(s) Resolved
This PR fixes a minor bug where a user could try to archive a discussion before they'd actually posted it, which caused an error. That button is now hidden until the comment is posted.

Since I was using that error to test out some monitoring improvements and the fix was small, I've combined some of the sentry changes into this PR:
- We now upload sourcemaps to sentry for the frontend and backend on each build
- The assert function was a no-op in prod, although it would still narrow the type, which made it easy to hide potential errors while using it! That's still true, but now we'll capture those failures in sentry.
- I removed the slow request warning since it's mostly noise at this point (and eating up our events quota)
- I made sure review instances get the commit hash (they don't have access to the heroku provided env var at run time, only build time) so they can set a release too

The only change in here that seems controversial/risky is that I've made our `wrap` function use async/await instead of `Promise.resolve` for compatibility with the node/v8 async tracing, [which only works with async/await and Promise.all](https://v8.dev/docs/stack-trace-api#async-stack-traces). Since this hasn't actually resolved the issue (stack traces still end at `process.processTicksAndRejections (node:internal/process/task_queues:95:5)`), I don't mind reverting this part, but it does seem like a step in the right direction.

## Test Plan
- Create a new discussion and observe that you cannot archive it until posting
- Make sure you can still archive it after comments are posted

## Screenshots (if applicable)

## Optional

### Notes/Context/Gotchas

### Supporting Docs
